src/util/prolog/prologResult.ts

```diff
@@ -51,70 +46,45 @@ export class PrologResult {
 
 	private getResultsFor(variable: string): string[] {
 		// Group 1 will be the content of the variable independent of the variable length.
-	 const pattern: any = new RegExp(variable + prologAnswerPattern, "g");
+	 const pattern = new RegExp(variable + variableAnswerPattern, "g");
 	 return this.regexGroupToArray(this.rawResults.toString(), pattern, 1);
 	}
 
 	private rawResultsToInnerArray(source: string): string[] {
 
-	if (source.includes("true")) { return ["true"]; }
-	if (source.includes("false")) { return["false"]; }
+	// Returns ["true"] or ["false"] in case of boolean answers
+	const booleanAnswer = booleanAnswerRegExp.exec(source);
+	if (booleanAnswer !== null) { return [booleanAnswer[1]]; }
 
-	const pattern: any = new RegExp(prologAnswerPattern, "g");
+	const pattern = new RegExp(variableAnswerPattern, "g");
 	return this.regexGroupToArray(source, pattern, 1);
 	}
 
 	private getVariables(): Set<string> {
-		const variables = new Set();
 
 		// Second group contains the variable names
-		const pattern: RegExp = /(^| )([A-Z]\w*)/g;
+		const pattern = /(^| )([A-Z]\w*)/g;
 		const rawVariables: string[] = this.regexGroupToArray(this.rawResults.toString(), pattern, 2);
 
-		rawVariables.forEach((e: string) => {
-			variables.add(e);
-		});
-	 return variables;
+		return new Set(rawVariables);
 	}
 
 	private getBooleans(): boolean[] {
-		const pattern: RegExp = /(true|false)(,| ;|.)/g;
-		let results: any[] = this.regexGroupToArray(this.rawResults.toString(), pattern, 1);
-		results = results.map((e: string) => {
-			return (e === "true") ? true : false;
-		});
+		const pattern = /(true|false)(,| ;|.)/g;
+		const results: string[] = this.regexGroupToArray(this.rawResults.toString(), pattern, 1);
+		const booleanResults: boolean[] = results.map((e: string) => e === "true");
```

Comment:
Nit: I don't think you strictly **need** to give an explicit type to bound variables in situations like this - but if it suits you, that is fine by me.